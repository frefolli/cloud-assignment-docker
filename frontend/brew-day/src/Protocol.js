export const ADVICE_ENDPOINT = '/api/advice';
export const BEERS_ENDPOINT = '/api/beers/';
export const BEER_LIST_ENDPOINT = '/api/beers';
export const DO_SHOPPING_ENDPOINT = '/api/shopping';
export const SHOPPING_ENDPOINT = '/api/shopping/';
export const INVENTORY_LIST_ENDPOINT = '/api/inventory';
export const INVENTORY_ENDPOINT = '/api/inventory/';
export const RECIPE_ENDPOINT = '/api/recipes/';
export const RECIPE_LIST_ENDPOINT = '/api/recipes';
export const RESET_ENDPOINT = '/api/reset';
export const SETTINGS_ENDPOINT = '/api/settings/';
export const SETTING_LIST_ENDPOINT = '/api/settings';
