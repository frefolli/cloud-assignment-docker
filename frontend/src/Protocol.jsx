import React from 'react';
import { Cookie, Feedback, Warning } from '@mui/icons-material';

export const ADVICE_ENDPOINT = '/api/advice';
export const BEERS_ENDPOINT = '/api/beers/';
export const BEER_LIST_ENDPOINT = '/api/beers';
export const DO_SHOPPING_ENDPOINT = '/api/shopping';
export const SHOPPING_ENDPOINT = '/api/shopping/';
export const INVENTORY_LIST_ENDPOINT = '/api/inventory';
export const INVENTORY_ENDPOINT = '/api/inventory/';
export const RECIPE_ENDPOINT = '/api/recipes/';
export const RECIPE_LIST_ENDPOINT = '/api/recipes';
export const RESET_ENDPOINT = '/api/reset';
export const SETTINGS_ENDPOINT = '/api/settings/';
export const SETTING_LIST_ENDPOINT = '/api/settings';

export const THEME_MANAGER_TRIGGER = "themeReload=true";
export const THEME_MANAGER_ESCAPE = "themeReload=false";
export const NAVBAR_THEME_MANAGER_TRIGGER = "navbarReload=true";
export const NAVBAR_THEME_MANAGER_ESCAPE = "navbarReload=false";
export const BACKGROUND_MANAGER_TRIGGER = "backgroundReload=true";
export const BACKGROUND_MANAGER_ESCAPE = "backgroundReload=false";

export const LAST_USED_THEME_LOCALSTORAGE_KEY = "lastUsedTheme";
export const LAST_USED_BACKGROUND_LOCALSTORAGE_KEY = "lastUsedBackground";

export const DEFAULT_THEME = "default";
export const DEFAULT_BACKGROUND = "default";

export const INGREDIENT_NAME_OPTIONS = ["acqua", "additivi", "lievito", "luppoli", "malto", "zuccheri"];

export const NOTE_TYPE_OPTIONS = ["generic", "taste", "problem"];
export const DEFAULT_NOTE_TYPE = "generic";
export const NOTE_TYPE_ICONS = {
    generic: <Feedback/>,
    taste: <Cookie/>,
    problem: <Warning/>
}