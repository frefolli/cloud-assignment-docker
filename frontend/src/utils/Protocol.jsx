import React from 'react';
import {Cookie, Feedback, Warning} from '@mui/icons-material';
import Notifier from './Notifier';

export const BACKEND_SERVER = 'http://brewday-backend';
export const ADVICE_ENDPOINT = BACKEND_SERVER + '/api/advice';
export const BEERS_ENDPOINT = BACKEND_SERVER + '/api/beers';
export const SHOPPING_ENDPOINT = BACKEND_SERVER + '/api/shopping';
export const INVENTORY_ENDPOINT = BACKEND_SERVER + '/api/inventory';
export const RECIPES_ENDPOINT = BACKEND_SERVER + '/api/recipes';
export const RESET_ENDPOINT = BACKEND_SERVER + '/api/reset';
export const SETTINGS_ENDPOINT = BACKEND_SERVER + '/api/settings';

export const ADVICE_VIEW_TRIGGER = 'adviceReload=true';
export const ADVICE_VIEW_ESCAPE = 'adviceReload=false';
export const NEXT_RECIPE_VIEW_TRIGGER = 'nextRecipeReload=true';
export const NEXT_RECIPE_VIEW_ESCAPE = 'nextRecipeReload=false';
export const THEME_MANAGER_TRIGGER = 'themeReload=true';
export const THEME_MANAGER_ESCAPE = 'themeReload=false';
export const NAVBAR_THEME_MANAGER_TRIGGER = 'navbarReload=true';
export const NAVBAR_THEME_MANAGER_ESCAPE = 'navbarReload=false';
export const BACKGROUND_MANAGER_TRIGGER = 'backgroundReload=true';
export const BACKGROUND_MANAGER_ESCAPE = 'backgroundReload=false';

export const LAST_USED_THEME_LOCALSTORAGE_KEY = 'lastUsedTheme';
export const LAST_USED_BACKGROUND_LOCALSTORAGE_KEY = 'lastUsedBackground';

export const DEFAULT_THEME = 'default';
export const DEFAULT_BACKGROUND = 'default';

export const INGREDIENT_NAME_OPTIONS = [
  'acqua', 'additivi', 'lievito', 'luppoli', 'malto', 'zuccheri'];

export const NOTE_TYPE_OPTIONS = ['generic', 'taste', 'problem'];
export const DEFAULT_NOTE_TYPE = 'generic';
export const NOTE_TYPE_ICONS = {
  generic: <Feedback/>,
  taste: <Cookie fontSize="large" sx={{verticalAlign: 'middle'}}/>,
  problem: <Warning/>,
};
export const FAKE_NOTIFIER = new Notifier(() => {});
export const isValidPositiveQuantity = (value) => (Number(value) > 0);
export const isNotValidPositiveQuantity = (value) =>
  (! isValidPositiveQuantity(value));
