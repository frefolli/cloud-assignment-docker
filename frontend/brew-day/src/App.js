
import React from 'react';
import { Route, Routes } from 'react-router';
import NavBar from './components/NavBar';
import Birre from './pages/Birre';
import Home from './pages/Home';
import Impostazioni from './pages/Impostazioni';
import Inventario from './pages/Inventario';
import Ricette from './pages/Ricette';
import Spesa from './pages/Spesa';

const App = () => {
  return (
    <React.Fragment>
      <NavBar />
      <Routes>
        <Route path="/" element={<Home />}/>
        <Route path="Ricette" element={<Ricette />}/>
        <Route path="Birre" element={<Birre />}/>
        <Route path="Inventario" element={<Inventario />}/>
        <Route path="Spesa" element={<Spesa />}/>                
        <Route path="Impostazioni" element={<Impostazioni />}/>
      </Routes>
    </React.Fragment>
  );
}

export default App;