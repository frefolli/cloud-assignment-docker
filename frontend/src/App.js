
import React from 'react';
import { Route, Routes } from 'react-router';
import NavBar from './components/NavBar';
import NavBarThemeManager from './components/NavBarThemeManager';
import Birre from './pages/Birre';
import Home from './pages/Home';
import Impostazioni from './pages/Impostazioni';
import Inventario from './pages/Inventario';
import Ricette from './pages/Ricette';
import Spesa from './pages/Spesa';

export default class App extends React.Component {
  render() {
    return (
      <React.Fragment>
        <NavBarThemeManager>
          <NavBar />
        </NavBarThemeManager>
        <Routes>
          <Route path="/" element={<Home />}/>
          <Route path="/Ricette" element={<Ricette />}/>
          <Route path="/Birre" element={<Birre />}/>
          <Route path="/Inventario" element={<Inventario />}/>
          <Route path="/Spesa" element={<Spesa />}/>                
          <Route path="/Impostazioni" element={<Impostazioni masterCall={() => this.setState({})}/>}/>
        </Routes>
      </React.Fragment>
    );
  };
}