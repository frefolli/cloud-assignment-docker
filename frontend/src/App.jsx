import React, { useState } from 'react';
import { Route, Routes } from 'react-router';
import NavBar from './components/NavBar';
import NavBarThemeManager from './components/NavBarThemeManager';
import Birre from './pages/Birre';
import Home from './pages/Home';
import Impostazioni from './pages/Impostazioni';
import Inventario from './pages/Inventario';
import Ricette from './pages/Ricette';
import Spesa from './pages/Spesa';
import { useSnackbar } from "notistack";

const App = () => {
  const [ flick, setFlick ] = useState(0);
  const { enqueueSnackbar } = useSnackbar();

  return (
      <React.Fragment>
        <NavBarThemeManager>
          <NavBar />
        </NavBarThemeManager>
        <Routes>
          <Route path="/" element={<Home />}/>
          <Route path="/Ricette" element={<Ricette enqueueSnackbar={enqueueSnackbar} />}/>
          <Route path="/Birre" element={<Birre />}/>
          <Route path="/Inventario" element={<Inventario />}/>
          <Route path="/Spesa" element={<Spesa />}/>                
          <Route path="/Impostazioni" element={<Impostazioni masterCall={() => setFlick({})}/>}/>
        </Routes>
      </React.Fragment>
  );
}

export default App;
