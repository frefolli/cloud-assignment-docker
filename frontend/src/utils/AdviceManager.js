import {ADVICE_ENDPOINT} from './Protocol';

export default class AdviceManager {
  getAdvice() {
    return new Promise((acc, rej) => {
      fetch(ADVICE_ENDPOINT)
          .then((res) => res.json())
          .then((data) => acc(data))
          .catch((err) => rej(err));
    });
  }
}
