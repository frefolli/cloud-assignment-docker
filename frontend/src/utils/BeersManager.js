import {BEERS_ENDPOINT} from './Protocol';

export default class BeersManager {
  getBeerList(search) {
    return new Promise((acc, rej) => {
      let url = BEERS_ENDPOINT;
      if (search !== undefined) {
        url += `?`;
        if (search.name !== '') {
          url += `&&name=${search.name}`;
        }
        if (search.recipe !== '') {
          url += `&&recipeID=${search.recipe}`;
        }
      }
      fetch(url, {mode: 'cors'})
        .then((response) => response.json())
        .then((data) => acc(data)).catch(rej);
    });
  }

  getBeer(beerID) {
    return new Promise((acc, rej) => {
      fetch(BEERS_ENDPOINT + `/${beerID}`, {mode: 'cors'})
        .then((response) => response.json())
        .then((data) => acc(data)).catch(rej);
    });
  }

  postBeer(beer) {
    return new Promise((acc, rej) => {
      fetch(BEERS_ENDPOINT, {
        method: 'POST',
        headers: {
          'Accept': 'application/json',
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(beer),
        mode: 'cors'
      }).
        then((data) => acc(data)).catch(rej);
    });
  }

  putBeer(beerID, beer) {
    return new Promise((acc, rej) => {
      fetch(BEERS_ENDPOINT + `/${beerID}`, {
        method: 'PUT',
        headers: {
          'Accept': 'application/json',
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(beer),
        mode: 'cors'
      }).
        then((data) => acc(data)).catch(rej);
    });
  }

  deleteBeer(beerID) {
    return new Promise((acc, rej) => {
      fetch(BEERS_ENDPOINT + `/${beerID}`, {
        method: 'DELETE',
        headers: {
          'Accept': 'application/json',
          'Content-Type': 'application/json',
        },
        mode: 'cors'
      })
        .then((data) => acc(data))
        .catch(rej);
    });
  }

  postNote(beerID, note) {
    return new Promise((acc, rej) => {
      fetch(BEERS_ENDPOINT + `/${beerID}`, {
        method: 'POST',
        headers: {
          'Accept': 'application/json',
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(note),
        mode: 'cors'
      }).
        then((data) => acc(data)).catch(rej);
    });
  }

  putNote(beerID, id, note) {
    return new Promise((acc, rej) => {
      fetch(BEERS_ENDPOINT + `/${beerID}/${id}`, {
        method: 'PUT',
        headers: {
          'Accept': 'application/json',
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(note),
        mode: 'cors'
      }).
        then((data) => acc(data)).catch(rej);
    });
  }

  deleteNote(beerID, id) {
    return new Promise((acc, rej) => {
      fetch(BEERS_ENDPOINT + `/${beerID}/${id}`, {
        method: 'DELETE',
        headers: {
          'Accept': 'application/json',
          'Content-Type': 'application/json',
        },
        mode: 'cors'
      })
        .then((data) => acc(data))
        .catch(rej);
    });
  }
}
