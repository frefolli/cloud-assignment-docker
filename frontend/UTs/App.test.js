import React from 'react';
import { render } from '@testing-library/react';
import App from '../src/App';
import reportWebVitals from '../src/reportWebVitals';
import { act } from "react-test-renderer";
import { SnackbarProvider } from 'notistack';
import { MemoryRouter} from "react-router-dom"
import { SETTINGS_ENDPOINT, SHOPPING_ENDPOINT } from '../src/utils/Protocol';

global.fetch = jest.fn().mockImplementation((url) =>
  Promise.resolve({
    json: () => {
      if (url.startsWith(SETTINGS_ENDPOINT + "/" + "equipment"))
        return Promise.resolve({value:"30"})
        if (url.startsWith(SETTINGS_ENDPOINT + "/"))
          return Promise.resolve({value:"default"})
      if (url.startsWith(SHOPPING_ENDPOINT + "/"))
        return Promise.resolve([]);
       return Promise.resolve({});
    },
  })
)

describe('App component', () => {
  test('should render correctly', async () => {
    let theContainer;
    await act(() => {
      const { container } = render(
        <MemoryRouter>
          <SnackbarProvider maxSnack={4}>
            <App />
          </SnackbarProvider>
        </MemoryRouter>
      );
      theContainer = container;
    })
    expect(theContainer.firstChild).toMatchSnapshot();
  })
});

describe('reportWebVitals', () => {
  test('should launch reportWebVitals', () => {
    reportWebVitals(() => true);
  });
});
