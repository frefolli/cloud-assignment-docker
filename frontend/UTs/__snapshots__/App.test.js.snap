// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render correctly 1`] = `
<header
  class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-1oiwwlu-MuiPaper-root-MuiAppBar-root"
>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1uzrjc8-MuiSvgIcon-root"
        data-testid="SportsBarIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 9h-1.56c.35-.59.56-1.27.56-2 0-2.21-1.79-4-4-4-.34 0-.66.05-.98.13-.82-.68-1.86-1.11-3.02-1.11-1.89 0-3.51 1.11-4.27 2.71C4.15 5.26 3 6.74 3 8.5c0 1.86 1.28 3.41 3 3.86V21h11v-2h2c1.1 0 2-.9 2-2v-6c0-1.1-.9-2-2-2zM7 10.5c-1.1 0-2-.9-2-2 0-.85.55-1.6 1.37-1.88l.8-.27.36-.76C8 4.62 8.94 4.02 10 4.02c.79 0 1.39.35 1.74.65l.78.65S13.16 5 13.99 5c1.1 0 2 .9 2 2h-3C9.67 7 9.15 10.5 7 10.5zM19 17h-2v-6h2v6z"
        />
      </svg>
      <a
        class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-m5x0z-MuiTypography-root"
        href="/"
      >
        BrewDay!
      </a>
      <div
        class="MuiBox-root css-1lvtzne"
      >
        <button
          aria-controls="menu-appbar"
          aria-haspopup="true"
          aria-label="OpenNavMenu"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="MenuIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-z5gn4u-MuiSvgIcon-root"
        data-testid="SportsBarIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 9h-1.56c.35-.59.56-1.27.56-2 0-2.21-1.79-4-4-4-.34 0-.66.05-.98.13-.82-.68-1.86-1.11-3.02-1.11-1.89 0-3.51 1.11-4.27 2.71C4.15 5.26 3 6.74 3 8.5c0 1.86 1.28 3.41 3 3.86V21h11v-2h2c1.1 0 2-.9 2-2v-6c0-1.1-.9-2-2-2zM7 10.5c-1.1 0-2-.9-2-2 0-.85.55-1.6 1.37-1.88l.8-.27.36-.76C8 4.62 8.94 4.02 10 4.02c.79 0 1.39.35 1.74.65l.78.65S13.16 5 13.99 5c1.1 0 2 .9 2 2h-3C9.67 7 9.15 10.5 7 10.5zM19 17h-2v-6h2v6z"
        />
      </svg>
      <a
        class="MuiTypography-root MuiTypography-h5 MuiTypography-noWrap css-1lfs2tq-MuiTypography-root"
        href="/"
      >
        BrewDay!
      </a>
      <div
        class="MuiBox-root css-1t6c9ts"
      >
        <a
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-18lfq1a-MuiButtonBase-root-MuiButton-root"
          href="/Ricette"
          tabindex="0"
        >
          Ricette
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
        <a
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-18lfq1a-MuiButtonBase-root-MuiButton-root"
          href="/Birre"
          tabindex="0"
        >
          Birre
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
        <a
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-18lfq1a-MuiButtonBase-root-MuiButton-root"
          href="/Inventario"
          tabindex="0"
        >
          Inventario
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
        <a
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-18lfq1a-MuiButtonBase-root-MuiButton-root"
          href="/Spesa"
          tabindex="0"
        >
          Spesa
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
        <a
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-18lfq1a-MuiButtonBase-root-MuiButton-root"
          href="/Impostazioni"
          tabindex="0"
        >
          Impostazioni
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
  </div>
</header>
`;
